lib/iris/coords.py

```diff
@@ -2083,7 +2090,8 @@ def cell(self, index):
         """
         index = iris.util._build_full_slice_given_keys(index, self.ndim)
 
-        point = tuple(np.array(self.core_points()[index], ndmin=1).flatten())
+        # Use `np.asanyaray` to preserve any masked values:
+        point = tuple(np.asanyarray(self.core_points()[index]).flatten())
```

Comment:
Can confirm that:

- `asanyarray()` preserves masked arrays
- While `asanyarray()` does not have the `ndmin` parameter, the final `flatten()` call prevents a scalar array anyway

👍

Reply:
Indeed. I did leave a comment to state this originally, but have since cleaned it up. Good that you've confirmed though!